lib/s3_helper.rb

```diff
@@ -260,7 +260,10 @@ def object(path)
   def self.s3_options(obj)
     opts = { region: obj.s3_region }
 
-    opts[:endpoint] = SiteSetting.s3_endpoint if SiteSetting.s3_endpoint.present?
+    if SiteSetting.s3_endpoint.present?
+      opts[:endpoint] = SiteSetting.s3_endpoint
+      opts[:force_path_style] = true
```

Comment:
This is required for minio. We cannot do local requests to `http://BUCKET.127.0.0.1`, it has to be `http://127.0.0.1/BUCKET`. This mostly works except for when we store the final Upload record, need to look into that more.